pymobiledevice3/remote/core_device_tunnel_service.py

```diff
@@ -0,0 +1,472 @@
+            {'type': PairingDataComponentType.IDENTIFIER, 'data': self.identifier.encode()},
+            {'type': PairingDataComponentType.PUBLIC_KEY,
+             'data': self.ed25519_private_key.public_key().public_bytes_raw()},
+            {'type': PairingDataComponentType.SIGNATURE, 'data': self.signature},
+            {'type': PairingDataComponentType.INFO, 'data': device_info},
+        ])
+
+        cip = ChaCha20Poly1305(setup_encryption_key)
+        encrypted_data = cip.encrypt(b'\x00\x00\x00\x00PS-Msg05', tlv, b'')
+
+        tlv = TLVBuf.build([
+            {'type': PairingDataComponentType.ENCRYPTED_DATA, 'data': encrypted_data[:255]},
+            {'type': PairingDataComponentType.ENCRYPTED_DATA, 'data': encrypted_data[255:]},
+            {'type': PairingDataComponentType.STATE, 'data': b'\x05'},
+        ])
+
+        response = self._send_receive_pairing_data({
+            'data': tlv,
+            'kind': 'setupManualPairing',
+            'sendingHost': 'User’s Mac mini',
+            'startNewSession': False})
+        data = self.decode_tlv(TLVBuf.parse(response))
+
+        tlv = TLVBuf.parse(cip.decrypt(
+            b'\x00\x00\x00\x00PS-Msg06', data[PairingDataComponentType.ENCRYPTED_DATA], b''))
+
+        return tlv
+
+    def _init_client_server_main_encryption_keys(self) -> None:
+        client_key = HKDF(
+            algorithm=hashes.SHA512(),
+            length=32,
+            salt=None,
+            info=b'ClientEncrypt-main',
+        ).derive(self.encryption_key)
+        self.client_cip = ChaCha20Poly1305(client_key)
+
+        server_key = HKDF(
+            algorithm=hashes.SHA512(),
+            length=32,
+            salt=None,
+            info=b'ServerEncrypt-main',
+        ).derive(self.encryption_key)
+        self.server_cip = ChaCha20Poly1305(server_key)
+
+    def _create_remote_unlock(self) -> None:
+        response = self._send_receive_encrypted_request({'request': {'_0': {'createRemoteUnlockKey': {}}}})
+        self.remote_unlock_host_key = response['createRemoteUnlockKey']['hostKey']
+
+    def _create_listener(self, private_key: RSAPrivateKey, protocol: str = 'quic') -> Mapping:
+        request = {'request': {'_0': {'createListener': {
+            'key': base64.b64encode(
+                private_key.public_key().public_bytes(Encoding.DER, PublicFormat.SubjectPublicKeyInfo)
+            ).decode(),
+            'transportProtocolType': protocol}}}}
+
+        response = self._send_receive_encrypted_request(request)
+        return response['createListener']
+
+    def _attempt_validate_pairing(self) -> None:
+        self.handshake_info = self._send_receive_handshake({'hostOptions': {'attemptPairVerify': True},
+                                                            'wireProtocolVersion': XpcInt64Type(19)})
+
+    def _validate_pairing(self) -> bool:
+        pairing_data = TLVBuf.build([
+            {'type': PairingDataComponentType.STATE, 'data': b'\x01'},
```

Comment:
What does state 1 mean?

Reply:
i don't really know it this is maybe the state we wish to move to, or the state we are currently at. i treat this now as just raw bytes without trying to figure out the deeper meaning right now